src/streamlink/plugins/dailymotion.py

```diff
@@ -1,37 +1,17 @@
-import json
 import re
 
-from functools import reduce
-
-from streamlink.compat import urlparse, range
 from streamlink.exceptions import NoStreamsError
 from streamlink.plugin import Plugin
 from streamlink.plugin.api import http, validate
-from streamlink.stream import HDSStream, HLSStream, HTTPStream, RTMPStream
-from streamlink.stream.playlist import FLVPlaylist
+from streamlink.stream import HLSStream, HTTPStream
 
 COOKIES = {
     "family_filter": "off",
     "ff": "off"
 }
-QUALITY_MAP = {
-    "ld": "240p",
-    "sd": "360p",
-    "hq": "480p",
-    "hd720": "720p",
-    "hd1080": "1080p",
-    "custom": "live",
-    "auto": "hds",
-    "source": "hds"
-}
-STREAM_INFO_URL = "https://www.dailymotion.com/sequence/full/{0}"
+STREAM_INFO_URL = "http://www.dailymotion.com/player/metadata/video/{0}"
```

Comment:
`https://` is available, maybe use it instead.

Reply:
Totally right, I've fixed it too